spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/data/RandomData.java

```diff
@@ -217,6 +218,11 @@ public Object map(Types.MapType map, Supplier<Object> keyResult, Supplier<Object
       return result;
     }
 
+    @Override
+    public Object variant(Types.VariantType variant) {
```

Comment:
What does this have to do with the changes to fixup types?

Reply:
Sure, I will implement it in the relevant PR.